include/fence.h

```diff
@@ -70,42 +70,34 @@ namespace celerity::detail {
 template <typename T>
 class host_object_fence_promise final : public detail::fence_promise {
   public:
-	explicit host_object_fence_promise(const experimental::host_object<T>& obj) : m_host_object(obj) {}
+	explicit host_object_fence_promise(const T* instance) : m_instance(instance) {}
```

Comment:
What's the background for this change?

Reply:
The fence promise is owned by the TDAG, so copying the host object here would lifetime-extend the instance, which we generally don't do anymore.